Client/Frontend/Browser/Browser.swift

```diff
@@ -327,7 +327,7 @@ private class HelperManager: NSObject, WKScriptMessageHandler {
     }
 
     func addHelper(helper: BrowserHelper, name: String) {
-        if let existingHelper = helpers[name] {
+        if let _ = helpers[name] {
```

Comment:
There are now 2 ways to check for nil optionals:

1. the good old if let
2. guard let a = statement else { do something; return }

the thing with guard is that nothing after the guard statement will be
executed in the else case, so if you have shared code for both cases after
the check for optional then you still should use if let

the _ is because we are not using the result of the if let in the
conditional block, in which case we should not create a proper variable for
it.

On 27 August 2015 at 17:51, Stephan Leroux <notifications@github.com> wrote:

> In Client/Frontend/Browser/Browser.swift
> <https://github.com/mozilla/firefox-ios/pull/966#discussion_r38118180>:
>
> > @@ -327,7 +327,7 @@ private class HelperManager: NSObject, WKScriptMessageHandler {
> >      }
> >
> >      func addHelper(helper: BrowserHelper, name: String) {
> > -        if let existingHelper = helpers[name] {
> > +        if let _ = helpers[name] {
>
> Question: Is this the preferred way now to check if a optional is not nil?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/firefox-ios/pull/966/files#r38118180>.
>